### PR TITLE
Added possibilities for custom sql expression:

### PR DESCRIPTION
**Pull request by OlegNadymov**
1. ordering by constant
 2. using constant as column access.

### CODE REVIEW

**Comment by OlegNadymov** [comment]
Could you give more details about failed tests, please?

**Comment by OlegNadymov** [comment]
@mythz could you help me ?

**Comment by OlegNadymov** [comment]
I've found problem. I'm working to fix it now.

**Comment by OlegNadymov** [comment]
I've fixed my bug in the PR.

**Comment by OlegNadymov** [comment]
Is something wrong with my PR?

File: src/ServiceStack.OrmLite/Expressions/SqlExpression.cs [hunk 1]
```diff
@@ -682,6 +683,12 @@ private SqlExpression<T> OrderByInternal(Expression keySelector)
             return this;
         }
 
+        private static bool ObjectIsClass(object obj)
+        {
+            return obj != null &&
+                   !(obj.GetType().IsPrimitive() || obj.GetType().IsValueType() || obj is string);
+        }
+
```
Comment (mythz) [inline src/ServiceStack.OrmLite/Expressions/SqlExpression.cs]: Why isn't this using `ob.GetType().IsClass()`?

Comment (OlegNadymov) [reply src/ServiceStack.OrmLite/Expressions/SqlExpression.cs]: I need to determine value types include ``string``. The ``ob.GetType().IsClass()`` is ``true`` for ``string``. Also I've found solution for this problem in your project in method ``ServiceStack.OrmLite.OrmLiteSPStatement.ConvertToScalarList<T>()``:
```
public List<T> ConvertToScalarList<T>()
        {
            if (!((typeof(T).IsPrimitive()) || typeof(T).IsValueType() || (typeof(T) == typeof(string)) || (typeof(T) == typeof(String))))
                throw new Exception("Type " + typeof(T).Name + " is a non primitive type. Use ConvertToList function.");
...
        }
```
So I can change like this:
```
        private static bool ObjectIsClass(object obj)
        {
            return obj != null &&
                   !(obj is string) &&
                   obj.GetType().IsClass();
        }
```
Is it necessary to do so?

Comment (mythz) [reply src/ServiceStack.OrmLite/Expressions/SqlExpression.cs]: IMO It makes more sense for the method called `ObjectIsClass()` to use the `IsClass()` API then using inverted checks of IsPrimitive + IsValueType.

But I'm not thrilled of the name of the method as it doesn't provide any indication of what it's guarding against, what other types is this method for apart from `PartialSqlString` ?

Comment (OlegNadymov) [reply src/ServiceStack.OrmLite/Expressions/SqlExpression.cs]: One more type is ``SelectList`` (``VisitNew(NewExpression nex``). So i've decided exclude all reference types.

Comment (mythz) [reply src/ServiceStack.OrmLite/Expressions/SqlExpression.cs]: It's more descriptive to list the concrete types so we know when this code-path is used.

Comment (OlegNadymov) [reply src/ServiceStack.OrmLite/Expressions/SqlExpression.cs]: Something like this?
```
        private static bool ObjectIsSqlClass(object obj)
        {
            return obj != null &&
                   (obj is PartialSqlString ||
                    obj is SelectList);
        }
```

Comment (mythz) [reply src/ServiceStack.OrmLite/Expressions/SqlExpression.cs]: Just IsSqlClass is fine.

**Comment by OlegNadymov** [comment]
Ok! I've done it.

**Comment by mythz** [comment]
Great, thx 👍

**Comment by OlegNadymov** [comment]
@mythz  Thanks for your advice!

**Pull request merged**